spec/support/shared_examples/provider_auth_shared_examples.rb

```diff
@@ -0,0 +1,6 @@
+RSpec.shared_examples 'a provider not authenticated' do
+  it 'should redirect the user to the login page' do
+    perform_request
+    expect(response).to redirect_to(new_provider_session_path)
+  end
```

Comment:
Two things:

1. If you pass the request into the shared example, you won't have to force the request to be wrapped as `perform_request`
2. If you log the user off before calling the request, you won't need to separate the logged on behaviour into a separate context. You'll be able to just have a general `before { login_as create(:provider) }` at the start of each spec.

So:
```ruby
RSpec.shared_examples 'a provider not authenticated' do |current_request|
   it 'should redirect the user to the login page' do
     sign_out :provider
     current_request
     expect(response).to redirect_to(new_provider_session_path)
  end
end
```
Then in the specs you call:
```ruby
it_behaves_like 'a provider not authenticated', perform_request
```